salt/orch/migration.sls

```diff
@@ -0,0 +1,85 @@
+{#- Get a list of nodes seem to be down or unresponsive #}
+{#- This sends a "are you still there?" message to all #}
+{#- the nodes and wait for a response, so it takes some time. #}
+
+{%- set parallelism = salt['pillar.get']('parallelism', 3) %}
+
+{%- set nodes_down = salt.saltutil.runner('manage.down') %}
+{%- if nodes_down|length >= 1 %}
+# {{ nodes_down|join(',') }} seem to be down: skipping
+  {%- do salt.caasp_log.debug('CaaS: nodes "%s" seem to be down: ignored', nodes_down|join(',')) %}
+  {%- set is_responsive_node_tgt = 'not L@' + nodes_down|join(',') %}
+{%- else %}
+# all nodes seem to be up
+  {%- do salt.caasp_log.debug('CaaS: all nodes seem to be up') %}
+  {#- we cannot leave this empty (it would produce many " and <empty>" targets) #}
+  {%- set is_responsive_node_tgt = '*' %}
+{%- endif %}
+
+{%- set is_migratable = is_responsive_node_tgt + ' and G@tx_update_migration_available:true' %}
+
+{%- set is_admin = is_migratable + ' and G@roles:admin' %}
+
+{%- set nodes_up = salt.saltutil.runner('manage.up')|reject("equalto", "ca") %}
```

Comment:
Having a `nodes_up` and a `nodes_down` is a waste of resources, as each `manage.up` involves sending a `ping` to all the machines in the cluster, waiting for the response, etc. You can build `nodes_up` as the _inverse_ of `nodes_down`...